src/lib/dialog/dialog-directives.ts

```diff
@@ -0,0 +1,49 @@
+import {Directive} from '@angular/core';
+import {MdDialog} from './dialog';
+
+
+/**
+ * Button that will close the current dialog.
+ */
+@Directive({
+  selector: 'button[md-dialog-close]',
+  host: {
+    '(click)': 'dialog.closeTop()'
```

Comment:
I went this route since it's not currently possible to have multiple dialogs side-by-side. Also I wasn't aware of the `DialogInjector`. I'll try it out.